client/payment-methods/index.js

```diff
@@ -73,6 +73,26 @@ const PaymentMethods = ( { enabledMethodIds, onEnabledMethodIdsChange } ) => {
 		);
 	};
 
+	const handleDragEnd = ( event ) => {
+		const { active, over } = event;
+
+		if ( active.id !== over.id ) {
+			const oldIndex = enabledMethodIds.indexOf( active.id );
+			const newIndex = enabledMethodIds.indexOf( over.id );
+
+			const enabledMethodIdsCopy = [ ...enabledMethodIds ];
+			enabledMethodIdsCopy.splice(
+				0 > newIndex
+					? enabledMethodIdsCopy.length + newIndex
+					: newIndex,
+				0,
+				enabledMethodIdsCopy.splice( oldIndex, 1 )[ 0 ]
+			);
```

Comment:
This is basically the equivalent of `arrayMove` from `@dnd-kit/sortable`, but I didn't import it here because I wanted to keep the `dnd-kit` dependencies in the `OrderableList` component.